rc/scdoc.kak

```diff
@@ -0,0 +1,54 @@
+# https://git.sr.ht/~sircmpwn/scdoc
+# ‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾
+
+# Detection
+# ‾‾‾‾‾‾‾‾‾
+
+hook global BufCreate .*[.]scd %{
+    set-option buffer filetype scdoc
+}
+
+# Initialization
+# ‾‾‾‾‾‾‾‾‾‾‾‾‾‾
+
+hook global WinSetOption filetype=scdoc %{
+    require-module scdoc
+    hook -once -always window WinSetOption filetype=.* %{ remove-hooks window scdoc-.+ }
+}
+
+hook -group scdoc-highlight global WinSetOption filetype=scdoc %{
+    add-highlighter window/scdoc ref scdoc
+    hook -once -always window WinSetOption filetype=.* %{ remove-highlighter window/scdoc}
+}
+
+
+provide-module scdoc %{
+
+# Highlighters
+# ‾‾‾‾‾‾‾‾‾‾‾‾
+
+add-highlighter shared/scdoc regions
+add-highlighter shared/scdoc/inline default-region regions
+add-highlighter shared/scdoc/inline/text default-region group
+
+add-highlighter shared/scdoc/inline/text/ regex \
+	'(?<!# )((?<!\\)|\\\\)\*(\n?[^\n])*?((?<!\\)|\\\\)\*' 0:bold
+add-highlighter shared/scdoc/inline/text/ regex \
+	'(?<!# )((?<!\\)|\\\\)\b_(\n?[^\n])*?((?<!\\)|\\\\)_\b' 0:italic
+# Let’s break this down:
+# (?<!# ) Tretead literally in headings
+# ((?<!\\)|\\\\) Treated literally after \ but not after \\
+# (\n?[^\n]) Line break is allowed but not a paragraph break
+
+# TODO: foo \\_bar_ is correctly highlighted as italic but
+#       foo\\_bar_ is wrongly highlighted as italic
+
+add-highlighter shared/scdoc/inline/text/ regex '^; [^\n]*' 0:comment
+add-highlighter shared/scdoc/inline/text/ regex '^##? [^\n]*' 0:header
+add-highlighter shared/scdoc/inline/text/ regex '\\.' 0:meta
+
+add-highlighter shared/scdoc/codeblock region -match-capture \
```

Comment:
I don't see any capture group in the opening regex, so `-match-capture` might not be necessary here?

Or maybe you meant to use `'(^\t*`{3}$)' '\1' fill meta`?

Reply:
I copied from markdown.kak again. `-match-capture` really seems to be not necessary. From doc-highlighters:
> If the -match-capture switch is passed, then region closing and recurse matches are considered valid for a given region opening match only if they
matched the same content for the capture 1.

Reply:
I edited my comment because of a formatting glitch. The code snippet shows what I think should be used.

I can make the changes myself in my own commit, c.f. the message above.

Reply:
I now understand what the purpose of `-match-capture` is! It does _not_ allow you to use captures like `\1` but makes sure that capture 1 of the start and the closing regex are identical. In our case it makes sure that the three backticks are on the same level of indentation, which is a good thing.

The code should be:

    region -match-capture (^\t?```$) (^\t?```$) fill meta

Reply:
Right, I think the `\1` thing used to be the way to implement that, and the documentation displayed when typing out `addhl window/ region ` in the prompt reinforced that idea (seems deprecated):

```
If -match-capture is specified, then regions end/recurse matches must have
the same \1 capture content as the begin match to be considered           
```